gradle-plugin/plugin/src/main/groovy/io/realm/gradle/Realm.groovy

```diff
@@ -1,13 +1,126 @@
         project.dependencies.add('compile', 'io.realm:realm-android:0.80.3') // TODO: make version dynamic
+
+        // Find all the variants
+        final def variants
+        if (isAndroidApp) {
+            variants = project.android.applicationVariants
+        } else {
+            variants = project.android.libraryVariants
+        }
+
+        // Do the bytecode manipulation
+        variants.all { variant ->
+            variant.dex.doFirst() {// Get the jar files of the imported libraries
+                JavaCompile javaCompile = variant.javaCompile
+                FileCollection classpathFileCollection = project.files(project.android.bootClasspath)
+                classpathFileCollection += javaCompile.classpath
+
+                // Initialize the class container for Javassist
+                ClassPool classPool = new ClassPool();
+
+                // Add the classpath jar files
+                classpathFileCollection.each {
+                    classPool.appendClassPath(it.toString())
+                }
+
+                // Append the folder containing the .class files for the project
+                classPool.appendClassPath(javaCompile.destinationDir.path)
+
+                // Find the class files
+                ConfigurableFileTree classFiles = project.fileTree(javaCompile.destinationDir);
+                classFiles.include("**/*.class");
+
+
+                def classCache = [:]
+
+                // Add custom accessors to the model classes
+                classFiles.each { classFile ->
+                    CtClass clazz = openClass(classFile, classPool)
+                    classCache[classFile] = clazz
+                    addRealmAccessors(clazz)
+                }
+
+                // Replace access fields with using the custom accessors
+                classFiles.each { classFile ->
+                    def clazz = classCache[classFile]
+                    useRealmAccessors(clazz as CtClass)
+                    (clazz as CtClass).writeFile(javaCompile.destinationDir.absolutePath)
+                }
+            }
+        }
+    }
+
+    private static void useRealmAccessors(CtClass clazz) {
+        clazz.getDeclaredBehaviors().each { behavior ->
+            if (!behavior.name.startsWith('realmGetter$') && !behavior.name.startsWith('realmSetter$')) {
+                behavior.instrument(new ExprEditor() {
+                    @Override
+                    void edit(FieldAccess fieldAccess) throws CannotCompileException {
+                        if (fieldAccess.field?.declaringClass?.superclass?.name?.equals("io.realm.RealmObject")) {
+                            println("Realm: Manipulating ${clazz.simpleName}.${behavior.name}(): ${fieldAccess.fieldName}")
+                            def fieldName = fieldAccess.fieldName
+                            if (fieldAccess.isReader()) {
+                                fieldAccess.replace('$_ = $0.realmGetter$' + fieldName + '();')
```

Comment:
Ok. It should be fine according to the Java Language Specification: http://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8